src/ir/field_selector_test.cc

```diff
@@ -0,0 +1,62 @@
+#include "src/ir/field_selector.h"
+
+#include <string>
+
+#include "absl/hash/hash_testing.h"
+#include "src/common/testing/gtest.h"
+
+namespace raksha::ir {
+
+static std::string sample_field_names[] = {
+    "foo",
+    "bar",
+    "foobar",
+    "baz",
+    "k",
+    "CapsCase",
+};
+
+class FieldSelectorTest : public ::testing::TestWithParam<std::string> {};
+
+TEST_P(FieldSelectorTest, ToStringJustDotPlusFieldName) {
+  const std::string field_name = GetParam();
+  FieldSelector field_selector(field_name);
+  const std::string field_selector_to_string = field_selector.ToString();
+  ASSERT_TRUE(field_selector_to_string.at(0) == '.');
+  ASSERT_TRUE(field_selector_to_string.substr(1) == field_name);
+}
+
+INSTANTIATE_TEST_SUITE_P(
+    FieldSelectorTestSuite,
+    FieldSelectorTest,
+    testing::ValuesIn(sample_field_names));
+
+TEST(FieldSelectorHashTest, FieldSelectorHashTest) {
+  std::vector<FieldSelector> field_selectors_to_check;
+  for (std::string field_name : sample_field_names) {
+    field_selectors_to_check.push_back(FieldSelector(field_name));
+  }
+  EXPECT_TRUE(
+      absl::VerifyTypeImplementsAbslHashCorrectly(field_selectors_to_check));
+}
+
+class FieldSelectorPairTest :
+   public ::testing::TestWithParam<::std::tuple<std::string, std::string>> {};
+
+TEST_P(FieldSelectorPairTest, FieldSelectorEqualsTest) {
```

Comment:
What is this testing exactly? Seems unnecessary?

Reply:
This tests that `FieldSelector`s are equal exactly when the field names from which they are built are equal.